### PR TITLE
Verticals popup menu on CourseUnitView

### PR DESCRIPTION
**Pull request by forgotvas**
Added verticals popup menu to CourseUnitView. It closed by feature key `VERTICALS_MENU_ENABLED` of `UI_COMPONENTS` category, to test it need to use config from [Config PR #157](https://github.com/edx/edx-mobile-config/pull/157).

This issue related to: [#156](https://github.com/openedx/openedx-app-ios/issues/156)

### Sample with key ###
https://github.com/openedx/openedx-app-ios/assets/480059/3667b4c1-4ebc-4c11-8153-668bf6c2dfd8

### Sample with disabled or without key ###
https://github.com/openedx/openedx-app-ios/assets/480059/cd3378b7-4f86-4006-8c83-0d4de5713f3d

### CODE REVIEW

**Comment by miankhalid** [comment]
@touchapp should we start the review of this PR or wait for:

1. product input on the linked issue i.e. #112?
2. and the scope and acceptance criteria to be added on the same issue?

File: Course/Course/Presentation/Unit/CourseUnitView.swift [hunk 1]
```diff
@@ -26,19 +26,45 @@ public struct CourseUnitView: View {
...
     private let sectionName: String
     public let playerStateSubject = CurrentValueSubject<VideoPlayerState?, Never>(nil)
     
+    //Dropdown parameters
+    @State var showDropdown: Bool = false
+    private let portraitTopSpacing: CGFloat = 60
+    private let landscapeTopSpacing: CGFloat = 75
+    
+    let isDropdownActive: Bool = {
+        Container.shared.resolve(ConfigProtocol.self)?.uiComponents.isVerticalsMenuEnabled ?? false
```
Comment (volodymyr-chekyrta) [inline Course/Course/Presentation/Unit/CourseUnitView.swift:40]: Would be better to pass this as an argument to the initializer or use ConfigProtocol from the viewModel

Comment (forgotvas) [reply Course/Course/Presentation/Unit/CourseUnitView.swift]: done

File: Course/Course/Presentation/Unit/CourseUnitViewModel.swift [hunk 2]
```diff
@@ -180,4 +180,21 @@ public class CourseUnitViewModel: ObservableObject {
     func trackFinishVerticalBackToOutlineClicked() {
         analytics.finishVerticalBackToOutlineClicked(courseId: courseID, courseName: courseName)
     }
+    
+    func route(to vertical: CourseVertical) {
+        if let index = verticals.firstIndex(where: {$0.id == vertical.id}),
```
[the remark this answers is not in the document]
Comment (forgotvas) [reply Course/Course/Presentation/Unit/CourseUnitViewModel.swift]: done

**Review by volodymyr-chekyrta — APPROVED** [review]

**Pull request merged**